test/run_test.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+import subprocess
+import sys
+
+TESTS = [
+    'autograd',
+    'cpp_extensions',
+    'cuda',
+    'dataloader',
+    'distributed',
+    'distributions',
+    'indexing',
+    'jit',
+    'legacy_nn',
+    'multiprocessing',
+    'nccl',
+    'nn',
+    'optim',
+    'sparse',
+    'torch',
+    'utils',
+]
+
+WINDOWS_BLACKLIST = [
+    'cpp_extensions',
+    'distributed',
+]
+
+
+def shell(command, cwd):
+    popen = subprocess.Popen(
+        command,
+        stdout=subprocess.PIPE,
+        universal_newlines=True,
+        cwd=cwd,
+        shell=True)
+    for stdout_line in iter(popen.stdout.readline, ''):
+        print(stdout_line.strip())
```

Comment:
Yes otherwise everything seemed to be double newlined. I can `strip('\n')`.

Not sure what you mean by "inherit". I just copied the code from https://stackoverflow.com/questions/4417546/constantly-print-subprocess-output-while-process-is-running 😛  Could you sketch out how I can improve it?

Reply:
Just don't specify the `stdout` argument. It should get inherited from the parent, and you won't have to do anything

Reply:
When you fork + exec and don't do anything special for stdout it will be inherited by the child just like any other file descriptor

Reply:
That doesn't seem to work unfortunately. It makes the output weirdly asynchronous. I imagine the pipe has something to do with the iterator? I'd rather not spend time right now on fixing something that works like we want

Reply:
That's weird. Can you only replace `.strip()` with `.strip('\n')`?

Reply:
Yes sorry I forgot that